Rewritten autoinstallation chapter of client configuration guide (#917)
uyuni-project/uyuni-docs: modules/client-configuration/pages/autoinst-distributions.adoc

[[autoinst-distributions]]
= Autoinstallation Distributions

> In order to offer a new distribution to install, you first need to have the files for the operating system you want to install on the clients. That includes the Linux kernel, an initial RAM disk, and other files required to boot the operating system in installation mode.

Author: "create" is wrong. I don't mean the technical steps to create an "autoinstallation distribution". I mean be able to offer a choice at installation.

Other changes are ok, thanks.

* you may extract these files from a DVD image - see xref:client-configuration:autoinst-distributions.adoc#based-on-iso-image[Distribution Based on an ISO Image];

> * you may also install [package]``tftpboot-installation`` - see xref:client-configuration:autoinst-distributions.adoc#based-on-rpm-package[Distribution Based on a RPM Package].

Author: Thanks, applied.

> You also need to have the matching base channel synchronized on your {productname} Server.

Author: "also" was the counterpart of "first". changed into "second".

> Then you might declare that distribution - see see xref:client-configuration:autoinst-distributions.adoc#declare-distribution[Declare an Autoinstallable Distribution].

Author: Almost technically correct (it's cobbler and not suse manager), and useful. it's what i meant by "offer" on top. But that was too allusive.

I'll move it on top as introduction on what we want to achieve.

> Then you might declare that distribution - see see xref:client-configuration:autoinst-distributions.adoc#declare-distribution[Declare an Autoinstallable Distribution].

Author: Registration is independant. It happens on the "profile" part.

> The resulting autoinstallation distribution will associate the files that you want to install to a base channel.

Author: This is the outcome of the whole process.

I hesitated to add that sentence. It kind of gives the technical side of what an "autoinstallation distribution" is, and was repeated by Michael several times during his presentation at the supporters training. On the other hand, I tried to remove as many "behind the scenes" details as possible. So I'm not sure we need it at all.

[[based-on-iso-image]]
== Distribution Based on an ISO Image

> This method assumes you have an installation media for the operating system you want to install on the clients. This is usually a DVD image that contains the Linux kernel, an [path]``initrd`` file, and other files required to boot the operating system in installation mode.

Author: ok.

Are you sure about the comma before "that" ? that sounds terribly German :stuck_out_tongue: .

> .Procedure: Import Files from an Installation Media

Author: 'k

> . Copy the installation media to your {productname} Server. For {suse} operating systems, you can download installation media from https://www.suse.com/download/.

Author: 'k

. Loop-mount the ISO image, and copy its contents somewhere:
----
# mount -o loop,ro some_image.iso /mnt
# mkdir -p /srv/www/distributions
# cp -a /mnt /srv/www/distributions/some_distribution

> # umount /mnt

Author: Right.

This is where I pefere DocBook, where you can use `<replaceable>` inside a line of code.

----

> . Take a note of this file path. You will use that information when you declare the distribution to {productname}.

Author: No, this is not where the image is stored, it is where the image files are extracted.

And I kind of prefer the original phrasing, as it refers to the operation previously explained.

> . Take a note of this file path. You will use that information when you declare the distribution to {productname}.

Author: This text is what was before my rewriting, I did not change it.

It also seems logical to me that 1) you copy the files somewhere 2) you take note of the path you chose. As separate steps, it makes sense.

[[based-on-rpm-package]]

> == Distribution Based on a RPM Package

Author: ok

> For {suse} systems, it may be easier to use prepackaged files for your installation system.

Author: OK, this was misleading. It was an introduction, not a warning. Will rephrase.

> .Procedure: Extract Files from an Installation Package

Author: yes

> . On the {productname} server, install the package [package]``tftpboot-installation``. You might determine its exact name with the command [command]``zypper se tftpboot-installation``. Determine where the installation files went with the command [command]``ls -d /usr/share/tftpboot-installation/*``.

Author: "If you need to" --> you will need to. Package name starts with `tftpboot-installation`, but the end is unknown.

I'll make that clearer.

And you will also necessarily need to determine that path, it's not an option.

> . Take a note of this file path. You will use that information when you declare the distribution to {productname}.

Author: no image involved here, we're speaking about a package

"this" better because it refers to instructions just given

as before, keeping that one

> . Take a note of this file path. You will use that information when you declare the distribution to {productname}.

Author: nope.

[NOTE]
====

> The above procedure will prepare the installation of the same operating system version as the one that powers your {productname} Server.

Author: yes

> If you want to install another version on the client, you will have to manually get the package [package]``tftpboot-installation`` from the distribution it belongs to.

Author: it's necessarily a different version, as this method works only for suse (as stated in the intro)

rest ok

> If you want to install another version on the client, you will have to manually get the package [package]``tftpboot-installation`` from the distribution it belongs to.

Author: well, strictly speaking you could want to change sles into opensuse. hmmm.

====

[[declare-distribution]]
== Declare an Autoinstallable Distribution

> .Procedure: Creating an Autoinstallable Distribution

Author: I see why you are suggesting it. It's the "create" button, but it's mostly a declarative task. ok.

* Select the matching [guimenu]``Base Channel``.
    This must match the installation media.
* Select the [guimenu]``Installer Generation``.
    This must match the installation media.
* OPTIONAL: Specify kernel options to use when booting this distribution.
    There are multiple ways to provide kernel options.
    Only add options here that are generic for the distribution.
. Click btn:[Create Autoinstallable Distribution].

Navigate to menu:Systems[Autoinstallation > Distributions] to manage your autoinstallable distributions.

[IMPORTANT]
====
The installation files that you prepared may, or may not, contain the packages to install. If they don't, add ``useonlinerepo=1`` to the [guimenu]``Kernel Options`` field.

The package repositories may contain signed or unsigned metadata. If the metadata are unsigned, add ``insecure=1`` to the [guimenu]``Kernel Options`` field, or use your own GPG key as explained in xref:client-configuration:autoinst-xxxxx.adoc[].

This is needed for example when you use the "online installer" ISO images instead of the full DVD, or when you use the [package]``tpboot-installation`` package.

> ====

Author: Much better indeed.

[NOTE]
====

> You can autoinstall {productname} Proxies in the same way as {sle} clients. Make sure you use the {sle} installation media, and choose the [guimenu]``SLE-Product-SUSE-Manager-Proxy-4.2-Pool for x86_64`` base channel.

Author: ok

When you have the files ready, and the base channel synchronized, you need to declare the distribution.
This operation associates the installation files to the base channel.
The distribution can be referred to by one or more installation profiles.
For more information, see xref:client-configuration:autoinst-distributions.adoc#declare-distribution[Declare an Autoinstallable Distribution].

[[based-on-iso-image]]
== Distribution Based on an ISO Image

This method assumes you have installation media for the operating system you want to install on the clients. 
This is usually a DVD [path]``.iso`` image that contains the Linux kernel, an [path]``initrd`` file, and other files required to boot the operating system in installation mode.

.Procedure: Importing Files from Installation Media
. Copy the installation media to your {productname} Server.
  For {suse} operating systems, you can download installation media from https://www.suse.com/download/.
. Loop-mount the ISO image, and copy its contents somewhere:
----
# mount -o loop,ro <image_name>.iso /mnt
# mkdir -p /srv/www/distributions
# cp -a /mnt /srv/www/distributions/<image_name>
# umount /mnt
----

>   Take a note of this file path. You will use that information when you declare the distribution to {productname}.

Reviewer: ```suggestion
  Take a note of the file path. 
  You will need it when you declare the distribution to {productname}.
```
